source/javascripts/controllers/awards_controller.coffee

```diff
@@ -21,16 +22,18 @@ EGA.controller "AwardsController", ($scope, $http, $sce, filterPipeline) ->
     {countries} = $scope.filters
     if countries.selected? then [countries.selected] else countries.names
 
-  filterAwards = -> $scope.filteredAwards = filterPipeline.filterFirstPage($scope.awards)
+  filterAwards = ->
+    $scope.filteredAwards = if $scope.awards then filterPipeline.filterFirstPage($scope.awards) else []
 
   initializeFilters = ->
-    $scope.filters.year.names = filter.helper.keys($scope.awards,'year').sort().reverse()
-    $scope.filters.category.names = ['earth', 'water', 'fire', 'air', 'youth', 'other']
+    if $scope.awards
+      $scope.filters.year.names = filter.helper.keys($scope.awards,'year').sort().reverse()
+      $scope.filters.category.names = ['earth', 'water', 'fire', 'air', 'youth', 'other']
 
-    for type in ['category', 'year']
-      $scope.filters[type].filterMap = filter.helper.trueMap($scope.filters[type].names)
+      for type in ['category', 'year']
+        $scope.filters[type].filterMap = filter.helper.trueMap($scope.filters[type].names)
 
-    $scope.filters.countries.names = filter.helper.keys($scope.awards, 'country').sort()
+      $scope.filters.countries.names = filter.helper.keys($scope.awards, 'country').sort()
```

Comment:
I don't like that `$scope.awards` is initially `undefined`. Now you've got to insert this conditional logic everywhere. If you leave the line

```coffeescript
$scope.awards = []
```

before

```coffeescript
$scope.preloadAwards = (awards) -> $scope.awards = awards
```

everything will work just fine.